src/transformers/training_args.py

```diff
@@ -1333,6 +1333,8 @@ def _setup_devices(self) -> "torch.device":
             device = torch.device("cuda", local_rank)
             self._n_gpu = 1
         elif is_sagemaker_dp_enabled():
+            import smdistributed.dataparallel.torch.torch_smddp  # noqa: F401
+
```

Comment:
i thought this was not needed anymore?

Reply:
this torch_smddp import is required to register smddp as a PT backend - the older version of smddp required a [different] import as well as other points of code changes which is no longer needed

Reply:
I see i guess this slipped through here: https://github.com/huggingface/transformers/pull/16371/files#diff-bfceaff300c851b8e24fc50dc6638482abaec8f7d2a718e877c3828c166bcf79